opentelemetry-api/src/opentelemetry/trace/__init__.py

```diff
@@ -237,8 +243,27 @@ def __exit__(
         exc_tb: typing.Optional[python_types.TracebackType],
     ) -> None:
         """Ends context manager and calls `end` on the `Span`."""
+
         self.end()
 
+        if (
+            exc_val is not None and
+            self.status is None and
+            self._auto_update_status
+        ):
+            if isinstance(exc_val, Status):
```

Comment:
It seems strange to have an exception derived from Status. Python enforces that all raised objects must derive from `BaseException`, otherwise a `TypeError` is raised. If we want to be able to set a status by throwing a special value, we should throw some exception type that *has* a status, not one that *is* a status. But I'd rather not support that since throwing an OpenTelemetry-specific exception type would mix observability and application logic and we definitely don't want an application's behavior to change depending on the used API implementation or observability configuration.

Reply:
Right, fixed that mistake. That being said, I acknowledge that the implementation as it is now is intended to allow the user to throw exceptions precisely in the way that you mention you don't want. I'm not trying to cause controversy here, is just that I feel like this implementation is what the issue submitter suggested, as OpenCensus does something similar :slightly_smiling_face: 

Again, this may or may not be the desired approach, please let me know in your comments about your thoughts on this :+1: